Content.Client/UserInterface/Controls/EntityListDisplay.cs

```diff
@@ -0,0 +1,302 @@
+        private bool _suppressScrollValueChanged;
+
+        public int ScrollSpeedY { get; set; } = 50;
+
+        private int ActualSeparation
+        {
+            get
+            {
+                if (TryGetStyleProperty(StylePropertySeparation, out int separation))
+                {
+                    return separation;
+                }
+
+                return SeparationOverride ?? DefaultSeparation;
+            }
+        }
+
+        public EntityListDisplay()
+        {
+            HorizontalExpand = true;
+            VerticalExpand = true;
+            RectClipContent = true;
+            MouseFilter = MouseFilterMode.Pass;
+
+            _vScrollBar = new VScrollBar
+            {
+                HorizontalExpand = false,
+                HorizontalAlignment = HAlignment.Right
+            };
+            AddChild(_vScrollBar);
+            _vScrollBar.OnValueChanged += ScrollValueChanged;
+        }
+
+        public void PopulateList(List<EntityUid> entities)
+        {
+            if (_count == 0 && entities.Count > 0)
+            {
+                EntityContainerButton control = new(entities[0]);
+                GenerateItem?.Invoke(entities[0], control);
+                control.Measure(Vector2.Infinity);
+                _itemHeight = control.DesiredSize.Y;
+                control.Dispose();
+            }
+            _count = entities.Count;
+            _entityUids = entities;
+            _updateChildren = true;
+            InvalidateArrange();
+        }
+
+        private void OnItemPressed(BaseButton.ButtonEventArgs args)
+        {
+            if (args.Button is not EntityContainerButton button)
+                return;
+            ItemPressed?.Invoke(button.EntityUid);
+        }
+
+        [Pure]
+        private Vector2 GetScrollValue()
+        {
+            var v = _vScrollBar.Value;
+            if (!_vScrollBar.Visible)
+            {
+                v = 0;
+            }
+            return new Vector2(0, v);
+        }
+
+        protected override Vector2 ArrangeOverride(Vector2 finalSize)
+        {
+            var separation = (int) (ActualSeparation * UIScale);
+
+            #region Scroll
+            var cHeight = _totalHeight;
+            var vBarSize = _vScrollBar.DesiredSize.X;
+            var (sWidth, sHeight) = finalSize;
+
+            try
+            {
+                // Suppress events to avoid weird recursion.
+                _suppressScrollValueChanged = true;
+
+                if (sHeight < cHeight)
+                    sWidth -= vBarSize;
+
+                if (sHeight < cHeight)
+                {
+                    _vScrollBar.Visible = true;
+                    _vScrollBar.Page = sHeight;
+                    _vScrollBar.MaxValue = cHeight;
+                }
+                else
+                    _vScrollBar.Visible = false;
+            }
+            finally
+            {
+                _suppressScrollValueChanged = false;
+            }
```

Comment:
wha

Reply:
Copy from ScrollContainer. Probably need @PJB3005's input on what it could be changed to.